src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/TestCases/FullStateDriverGenerator/UseBoolArrayArg.cpp

```diff
@@ -0,0 +1,112 @@
+//------------------------------------------------------------------------------
+// This code was generated by a tool.
+// <auto-generated />
+//------------------------------------------------------------------------------
+
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "CLI11.hpp"
+
+#include "QirRuntime.hpp"
+#include "QirContext.hpp"
+
+#include "SimFactory.hpp"
+
+using namespace Microsoft::Quantum;
+using namespace std;
+
+// Auxiliary functions for interop with Q# Array type.
+struct InteropArray
+{
+    int64_t Size;
+    void* Data;
+
+    InteropArray(int64_t size, void* data) :
+        Size(size),
+        Data(data){}
+};
+
+template<typename T>
+unique_ptr<InteropArray> CreateInteropArray(vector<T>& v)
+{
+    unique_ptr<InteropArray> array(new InteropArray(v.size(), v.data()));
+    return array;
+}
+
+template<typename S, typename D>
+void TranslateVector(vector<S>& sourceVector, vector<D>& destinationVector, function<D(S&)> translationFunction)
```

Comment:
Not used, but I assume it's part of the template that generates this code.

Reply:
Yes, this one is not used for some cases of array data types but if we handle all the specific data types individually to not include this, the template code becomes more difficult to read.

One of the future things to do is to extract these functions into a header-only library to make the generated file smaller.